contrib/tester-progs/fork-tester.c

```diff
@@ -33,6 +33,12 @@ void child1()
 			int ppid = getppid();
 			if (ppid == 1) {
 				break;
+			} else {
+				// If systemd run with --user
+				const char *manager_pid_str = getenv("MANAGERPID");
+				int manager_pid = atoi(manager_pid_str);
```

Comment:
I thought about this, but since `atoi` can receive a NULL pointer, I was thinking to keep it basic this way. If `atoi` receive a NULL pointer or a string that could not be transformed to an integer, it returns 0 which is an invalid PID anyway.
But if you prefer the very explicit way, I have no problem adding those checks, what do you think ?